internal/pkg/deploy/cloudformation/app.go

```diff
@@ -231,8 +231,8 @@ func (cf CloudFormation) getResourcesForStackInstances(app *config.Application,
 // AddServiceToApp attempts to add new service specific resources to the application resource stack.
 // Currently, this means that we'll set up an ECR repo with a policy for all envs to be able
 // to pull from it.
-func (cf CloudFormation) AddServiceToApp(app *config.Application, svcName string) error {
-	if err := cf.addWorkloadToApp(app, svcName); err != nil {
+func (cf CloudFormation) AddServiceToApp(app *config.Application, svcName string, createECR bool) error {
+	if err := cf.addWorkloadToApp(app, svcName, createECR); err != nil {
```

Comment:
What about instead of `createECR` we do this:
```go
type AddServiceToAppOpt func(*stack.AppResourcesService)

func WithoutECR(config *stack.AppResourcesService) { config.WithECR = false }

func (cf CloudFormation) AddServiceToApp(app *config.Application, svcName string, opts ...AddServiceToAppOpt) error
```

I know there are no other `opts` so far, but I feel like it isn't too early to establish a more scalable pattern now either. In the future, if we need to add more `opts` func to `addServiceToApp`, it'd be easy for the (potentially new) developers to just copy the pattern instead of keep adding boolean.

Reply:
Addressed!